thinc/model.py

```diff
@@ -285,7 +285,7 @@ def set_ref(self, name: str, value: Optional["Model"]) -> None:
         else:
             raise ValueError("Cannot add reference to node not in tree.")
 
-    def __call__(self, X: InT, is_train: bool) -> Tuple[OutT, Callable]:
+    def __call__(self, X: InT, is_train: bool) -> Tuple[OutT, Callable[[OutT], InT]]:
```

Comment:
We can't universally commit to this signature. For instance, tok2vec models have a signature like

```
Model[List[Doc], List[Floats2d]]
```

But you cannot return gradients of the type `IntT`, which is `List[Doc]` in that case.

Reply:
I see! Thanks for pointing that out. I'll close this PR :)